packages/app-builder/src/components/Scenario/AstBuilder/AstBuilderNode/Operand/OperandEditor/OperandEditModal/IsRoundedEdit/IsRoundedEdit.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+  type AstNode,
+  type IsRoundedAstNode,
+  NewConstantAstNode,
+  NewIsRoundedAstNode,
+} from '@app-builder/models';
+import { type AstNodeErrors } from '@app-builder/services/validation/ast-node-validation';
+import { formatNumber, useFormatLanguage } from '@app-builder/utils/format';
+import { useTranslation } from 'react-i18next';
+import { Button, ModalV2, Select } from 'ui-design-system';
+
+import { useIsRoundedEditState } from './IsRoundedEdit.hook';
+import { LeftOperand } from './LeftOperand';
+
+const thresholdOptions = [
+  { value: '1' },
+  { value: '10' },
+  { value: '100' },
+  { value: '1000' },
+  { value: '10000' },
+  { value: '100000' },
+  { value: '1000000' },
+  { value: '10000000' },
+  { value: '100000000' },
+  { value: '1000000000' },
+];
+
+export function IsRoundedEdit({
+  initialIsRoundedAstNode,
+  initialAstNodeErrors,
+  onSave,
+}: {
+  initialIsRoundedAstNode: IsRoundedAstNode;
+  initialAstNodeErrors: AstNodeErrors;
+  onSave: (astNode: AstNode) => void;
+}) {
+  const { t } = useTranslation(['scenarios', 'common']);
+  const language = useFormatLanguage();
+
+  const { value, setValue, threshold, setThreshold } = useIsRoundedEditState(
+    initialIsRoundedAstNode,
+    initialAstNodeErrors,
+  );
+
+  const handleSave = () => {
+    const isRoundedAstNode = NewIsRoundedAstNode(
+      value.astNode,
+      NewConstantAstNode({ constant: threshold.value }),
+    );
+    onSave(isRoundedAstNode);
+  };
+
+  return (
+    <>
+      <ModalV2.Title>{t('scenarios:edit_is_rounded.title')}</ModalV2.Title>
+      <div className="flex flex-col gap-9 p-6">
+        <div className="flex flex-col gap-2">
+          <div className="flex gap-2">
+            <LeftOperand
+              astNode={value.astNode}
+              astNodeErrors={value.astNodeErrors}
+              validationStatus={
+                (value.astNodeErrors?.errors ?? []).length > 0
+                  ? 'error'
+                  : 'valid'
+              }
+              onChange={setValue}
+            />
+            <div className="border-grey-10 bg-grey-02 flex h-10 w-fit min-w-[40px] items-center justify-center rounded border p-2 text-center">
+              <span className="text-s text-grey-100 font-medium">
+                is multiple of
+              </span>
+            </div>
+            <Select.Default
+              value={threshold.value.toString()}
+              onValueChange={(value) => setThreshold(parseInt(value, 10))}
+            >
+              {thresholdOptions.map((thresholdOption) => (
+                <Select.DefaultItem
+                  key={thresholdOption.value}
+                  value={thresholdOption.value}
+                >
+                  {formatNumber(parseInt(thresholdOption.value, 10), {
+                    language,
+                    style: undefined,
+                  })}
+                </Select.DefaultItem>
+              ))}
+            </Select.Default>
```

Comment:
I would change the `thresholdOptions` definition to take numbers, so it is closer to what we have for `threshold.value`
```suggestion
            <Select.Default
              value={threshold.value.toString()}
              onValueChange={(value) => setThreshold(parseInt(value, 10))}
            >
              {thresholdOptions.map((thresholdOption) => (
                <Select.DefaultItem
                  key={thresholdOption.value}
                  value={thresholdOption.value.toString()}
                >
                  {formatNumber(thresholdOption.value, {
                    language,
                    style: undefined,
                  })}
                </Select.DefaultItem>
              ))}
            </Select.Default>
```